src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -307,11 +307,16 @@ public Task<Compilation> GetCompilationAsync(SolutionState solution, Cancellatio
                 }
                 else
                 {
-                    return GetOrBuildCompilationInfoAsync(solution, lockGate: true, cancellationToken: cancellationToken)
-                        .ContinueWith(t => t.Result.Compilation, cancellationToken, TaskContinuationOptions.ExecuteSynchronously | TaskContinuationOptions.OnlyOnRanToCompletion, TaskScheduler.Default);
+                    return GetCompilationSlowAsync(solution, cancellationToken);
```

Comment:
Can we check if this pattern exists elsewhere?  It might be sufficient to just look for cases of .ExecuteSynchronously to see if we copy-pasted this elsewhere.